crates/rspack_tasks/src/lib.rs

```diff
@@ -0,0 +1,97 @@
+// borrow the ideas from turbo_tasks https://github.com/vercel/next.js/blob/678ef8b5650871a730ca14c480c762ca53716575/turbopack/crates/turbo-tasks/src/manager.rs#L1
+// which creates a implicit compiler context to support isolated parallel compiler state
+use std::{
+  future::Future,
+  sync::{atomic::AtomicU32, Arc},
+};
+
+use tokio::{
+  task::{futures::TaskLocalFuture, JoinHandle},
+  task_local,
+};
+
+// don't overuse this and put everything here, it's mostly used for store isolated id generator
+#[derive(Debug)]
+pub struct CompilerContext {
+  dependenc_id_generator: AtomicU32,
```

Comment:
I don't like this encapsulation. rspack_tasks should be similar to a toolkit, encapsulating task local related functions. compiler_context should be what rspack_core needs to pay attention to.

Reply:
what's your suggest change?

Reply:
just expose within_context and get_context.
``` rust
// rspack_core
use rspack_tasks::{within_context, get_context};

struct CompilerContext {
   dependency_id_generate: AtomicU32
}

fn build() {
   within_context(self.compiler_context, async { ... })
}

impl EntryDependency {
  pub fn new() {
    get_context::<CompilerContext>().fetch_new()
  }
}

```

Reply:
I'm more concerned with the implement behind this `define_context` macro, which introduces some lookup overhead anyway. actually i don't think a macro is needed here.

Reply:
I personally don't think rspack_tasks is a generic toolkit, it's more like a specific core data structure which only works for compiler context, so I'm not sure `define_context` will bring benifits here

Reply:
And I'm open  with change the crate name to more specific name like `rspack_comiler_tasks`, It's just that I'm ok with current's naming

Reply:
Another small question, if debug_context is to be implemented later, the ModuleGraph type may need to be imported from rspack_core into compilerContext. How to solve this package circular reference? Use pointers directly?

Reply:
yeah just use dark magic, it's just my random thoughts, if it's causing problem, I'm totoally ok with remove it